### PR TITLE
Bump version to 2.1.0.0

### PR DESCRIPTION
**Pull request by cliu123**
Signed-off-by: cliu123 <lc12251109@gmail.com>

### Description
* Category (Enhancement, New feature, Bug fix, Test fix, Refactoring, Maintenance, Documentation) Maintenance
* Bump security plugin version to 2.1.0.0 and OpenSearch core version to 2.1.0.
* Upgrading Gradle to [the version that OpenSearch core uses.](https://github.com/opensearch-project/OpenSearch/blob/main/gradle/wrapper/gradle-wrapper.properties#L14)
```
* What went wrong:
A problem occurred evaluating root project 'bwc-test'.
> Failed to apply plugin class 'org.opensearch.gradle.info.GlobalBuildInfoPlugin'.
   > Gradle 7.4.1+ is required
```

### Testing
UTs

### Check List
- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: bwc-test/build.gradle [hunk 1]
```diff
@@ -76,13 +76,13 @@ dependencies {
...
 String baseName = "securityBwcCluster"
 String bwcFilePath = "src/test/resources/"
-String projectVersion = "2.0.0.0"
+String projectVersion = "2.1.0.0"
 
 2.times {i ->
     testClusters {
         "${baseName}$i" {
             testDistribution = "ARCHIVE"
-        versions = ["1.3.0","2.0.0"]
```
Comment (peternied) [inline bwc-test/build.gradle:85]: Should these versions be additive instead of replacing the previous version?

File: bwc-test/build.gradle [hunk 2]
```diff
@@ -76,13 +76,13 @@ dependencies {
 String bwcVersion = "1.3.0.0";
 String baseName = "securityBwcCluster"
 String bwcFilePath = "src/test/resources/"
-String projectVersion = "2.0.0.0"
```
Comment (peternied) [inline bwc-test/build.gradle:79]: Can we pull this from the other gradle configuration? so we don't have to keep updating both files?

File: gradle/wrapper/gradle-wrapper.properties [hunk 3]
```diff
@@ -1,5 +1,5 @@
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-7.3.3-bin.zip
```
Comment (peternied) [inline gradle/wrapper/gradle-wrapper.properties:3]: Why the gradle upgrade, is it required? (I don't see details in the description about this)

Comment (cliu123) [reply gradle/wrapper/gradle-wrapper.properties:3]: added the details to the PR description

File: bwc-test/build.gradle (hunk 1 above)
Comment (cliu123) [reply bwc-test/build.gradle:85]: Upgrading to `2.0.0` is tested on `2.0` branch.

File: bwc-test/build.gradle (hunk 2 above)
Comment (cliu123) [reply bwc-test/build.gradle:79]: This is the only occurrence. We don't have to pull it to other places.

File: bwc-test/build.gradle (hunk 1 above)
Comment (peternied) [reply bwc-test/build.gradle:85]: My concern is if we add an incompatibility in version in 2.1 that doesn't work with 2.0.  Can you help me find where we test this scenario?

File: bwc-test/build.gradle (hunk 2 above)
Comment (peternied) [reply bwc-test/build.gradle:79]: > This is the only occurrence.

How is this the case? This pull request has effectively the same change at build.gradle@60, bwc-test/build.gradle@50 and bwc-test/build.gradle@79

Comment (cliu123) [reply bwc-test/build.gradle:79]: I see what you're referring to now. I thought that you were referring to this line#79 specifically. This is security plugin version. The others are OpenSearch version.
The 2 files are in 2 project folders. `bwc-test` is a sub-project of security plugin project. I don't see how the versions in 2 different project can be extracted to a central place though.

File: bwc-test/build.gradle (hunk 1 above)
Comment (cliu123) [reply bwc-test/build.gradle:85]: That's a great point! The test should validate the upgrade from 2.0.0 to 2.1.0.

Comment (peternied) [reply bwc-test/build.gradle:85]: According to our support matrix, we support the last major and the current previous major, so we should have tests for `1.3 -> 2.1` and `2.0 -> 2.1`, can we make sure we test both?

stated differently, we should have tests that cover: 
* last supported major version (1) and its most recent minor version (3) to current version
* current major version (2) and its most recent minor version (0) to current version 

if we really wanted to be thorough I think we would add checks for every minor version on the current major version (e.g. if we were on 2.4, we would have BWC tests for 1.3, 2.0, 2.1, 2.2, and 2.3) (e.g.2 if we were on 3.1 we would have BWC tests for 2.3 and 3.0)

I think we should codify your preferred plan for this is in a comment side by side with these values to make future changes clear.

File: bwc-test/build.gradle (hunk 2 above)
Comment (peternied) [reply bwc-test/build.gradle:79]: Let's table this for after we've correctly rewritten the build.gradle https://github.com/opensearch-project/security/issues/1839

File: bwc-test/build.gradle (hunk 1 above)
Comment (cliu123) [reply bwc-test/build.gradle:85]: >the current previous major

Do you mean current previous minor?

> According to our support matrix, we support the last major and the current previous major, so we should have tests for 1.3 -> 2.1 and 2.0 -> 2.1

Could you post the link to the support matrix? I don't see any plugins testing upgrade from multiple previous versions to the current version though.

Comment (cliu123) [reply bwc-test/build.gradle:85]: the current previous major

Do you mean current previous minor?

According to our support matrix, we support the last major and the current previous major, so we should have tests for 1.3 -> 2.1 and 2.0 -> 2.1

Could you post the link to the support matrix? We should follow if there is a consistent requirements, but I don't see any plugins testing upgrade from multiple previous versions to the current version though.

Comment (peternied) [reply bwc-test/build.gradle:85]: @cliu123 I've created https://github.com/opensearch-project/OpenSearch/issues/3440 to follow up for broad guidance.  Lets move ahead with the work you have done here.

**Review by DarshitChanpura — APPROVED** [review]

**Pull request merged**